prow/pjutil/pprof.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package pjutil contains helpers for working with ProwJobs.
+package pjutil
+
+import (
+	"net/http"
+	"net/http/pprof"
+
+	"github.com/sirupsen/logrus"
+)
+
+// ServePProf sets up a handler for pprof debug endpoints and starts a server for them asynchronously.
+func ServePProf() {
+	pprofMux := http.NewServeMux()
+	pprofMux.HandleFunc("/debug/pprof/", pprof.Index)
+	pprofMux.HandleFunc("/debug/pprof/cmdline", pprof.Cmdline)
+	pprofMux.HandleFunc("/debug/pprof/profile", pprof.Profile)
+	pprofMux.HandleFunc("/debug/pprof/symbol", pprof.Symbol)
+	pprofMux.HandleFunc("/debug/pprof/trace", pprof.Trace)
```

Comment:
I think `pprof.Index` already handles these subpaths given that `/debug/pprof/heap` already works with Deck.

https://golang.org/pkg/net/http/pprof/#Index and https://golang.org/src/net/http/pprof/pprof.go#L229 confirm this.

Reply:
I thought so too... but then without the other handlers I get

```
$ go tool pprof -http=: localhost:6060/debug/pprof/profile
Fetching profile over HTTP from http://localhost:6060/debug/pprof/profile
localhost:6060/debug/pprof/profile: server response: 404 Not Found - Unknown profile
failed to fetch any source profiles
```